Source/Patches.cs

```diff
@@ -542,4 +1044,117 @@ public static void Postfix()
 			}
 		}
 	}
-}
\ No newline at end of file
+
+	[HarmonyPatch(typeof(UIRoot), nameof(UIRoot.UIRootUpdate))]
+	public static class UIRoot_UIRootUpdate_OpinionReport_Patch
+	{
+		static int opinionTicks = -1;
+		private static Dictionary<(Pawn, Pawn), int> lastOpinions = new Dictionary<(Pawn, Pawn), int>();
+
+		public static void Postfix()
+		{
+			try
+			{
+				var map = Find.CurrentMap;
+				if (map == null) return;
+
+				opinionTicks++;
+
+				if (opinionTicks % 12000 == 0)
+				{
+					List<string> opinionMessages = new List<string>();
+					var freeColonistsCopy = map.mapPawns.FreeColonists.ToList();
+
+					foreach (var colonist in freeColonistsCopy)
+					{
+						foreach (var otherColonist in freeColonistsCopy)
+						{
+							if (colonist != otherColonist)
+							{
+								int currentOpinion = colonist.relations.OpinionOf(otherColonist);
+
+								if (lastOpinions.TryGetValue((colonist, otherColonist), out int previousOpinion))
+								{
+									string trend = currentOpinion == previousOpinion ? "stayed the same" :
+												   currentOpinion > previousOpinion ? "improved" : "worsened";
+
+									opinionMessages.Add($"{colonist.Name} current opinion of {otherColonist.Name}: {currentOpinion} (has {trend} since yesterday)");
```

Comment:
Isn't this a good example of patching the moment where a colonist gets or changes the opinion of someone? That would avoid polling.

Reply:
We do patch the moment when the change occurs, but I also think it valuable to get a periodic state of overall opinions - I could be wrong though.

Reply:
Sure, I see the value